jekyll/_cci2/optimization-cookbook.md

```diff
@@ -0,0 +1,153 @@
+
+**Note**For more detailed information about the CircleCI Queueing orb, refer to the following CircleCI pages:
+- Queueing and Single Threading Overview - https://github.com/eddiewebb/circleci-queue
+- CircleCI Queueing Orb - https://circleci.com/orbs/registry/orb/eddiewebb/queue#quick-start
+
+### Setting Up and Configuring Your Environment to use the CircleCI Platform and CircleCI Orbs
+
+To configure your environment for the CircleCI platform and CircleCI orbs, follow the steps listed below.
+
+1) Use CircleCI `version 2.1` at the top of your `.circleci/config.yml` file.
+
+`version: 2.1`
+
+2) If you do not already have Pipelines enabled, you'll need to go to **Project Settings -> Advanced Settings** to enable pipelines.
+
+3) Add the `orbs` stanza below your version, invoking the orb:
+
+`orbs:
+  queue: eddiewebb/queue@1.1.2`
+
+4) Use `queue` elements in your existing workflows and jobs.
+
+5) Opt-in to use of third-party orbs on your organization’s **Security Settings** page.
+
+### Blocking Workflows
+
+One of the easiest ways to prevent workflow concurrency using the CircleCI Single-Threading orb is to enable "blocking" of any workflows with an earlier timestamp. By setting the `block-workflow` parameter value to `true`, all workflows will be forced to run consecutively, not concurrently, thereby limiting the number of workflows in the queue. In turn, this will also improve overall performance while making sure no workflows are discarded.
+
+{% raw %}
+
+```yaml
+Version: 2.1
+docker:
+  - image: 'circleci/node:10'
+parameters:
+  block-workflow:
+    default: true
+    description: >-
+      If true, this job will block until no other workflows with an earlier
+      timestamp are running. Typically used as first job.
+    type: boolean
+  consider-branch:
+    default: true
+    description: Should we only consider jobs running on the same branch?
+    type: boolean
+  consider-job:
+    default: false
+    description: Deprecated. Please see block-workflow.
+    type: boolean
+  dont-quit:
+    default: false
+    description: >-
+      Quitting is for losers. Force job through once time expires instead of
+      failing.
+    type: boolean
+  only-on-branch:
+    default: '*'
+    description: Only queue on specified branch
+    type: string
+  time:
+    default: '10'
+    description: How long to wait before giving up.
+    type: string
+  vcs-type:
+    default: github
+    description: Override VCS to 'bitbucket' if needed.
+    type: string
+steps:
+  - until_front_of_line:
+      consider-branch: <<parameters.consider-branch>>
+      consider-job: <<parameters.consider-job>>
+      dont-quit: <<parameters.dont-quit>>
+      only-on-branch: <<parameters.only-on-branch>>
+      time: <<parameters.time>>
+      vcs-type: <<parameters.vcs-type>>
+```
+{% endraw %}
+
+## Caching
```

Comment:
Caching what exactly? "Caching dependencies" might be a more suitable title for this section. IMO the more specific the recipe the better. "Fried Brussel sprouts" vs "Fried food": which one sounds more appetizing/relevant to you?

Reply:
fixed in document.